src/utils/authFetch.js

```diff
@@ -0,0 +1,19 @@
+import {jwtUtils} from "./jwt/jwtUtils";
+import {loginAPI} from "./apis/LoginAPI";
+
+export const authFetch = (url, data) => {
+    const authToken = jwtUtils.getAuthToken()
+    const dataHeaders = data?.headers
+    const authHeaders = authToken ? {[loginAPI.jwtTokenKey]: authToken} : {}
+    const newHeaders = dataHeaders ? Object.assign(dataHeaders, authHeaders) : authHeaders
+    const authData = Object.assign(data | {}, {headers: newHeaders})
+    return fetch(url, authData).then(response => {
+        if(response.status >= 200 && response.status < 300)
+            return response
+        else
+            return jwtUtils.refreshToken().then(() => {
+                authData.headers[loginAPI.jwtTokenKey] = jwtUtils.getAuthToken()
+                return fetch(url, authData)
+            })
```

Comment:
Co, jak będzie niepoprawny token? w sensie walnę sobie "abc" - nie wpadnie to w jakąś rekurencję?

Reply:
Plus pewnie przydałoby się to jakkolwiek złapać - nawet jeśli walnie tylko w konsoli, wolałbym, żeby osoby, które konsolę oglądają (wszyscy wiemy, o kogo chodzi), nie widziały błędów z fetcha. xDD

Reply:
Czemu miałaby tu być rekurencja xD

Reply:
Nazwy funkcji mi się powaliły, w takim razie nie ważne XD